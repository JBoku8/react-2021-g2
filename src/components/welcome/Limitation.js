import Button from '../ui/Button';

function Limitation({ onClick, message }) {
  return (
    <div className="row flex-column">
      <h3 className="text-danger col-4 mt-3">{message}</h3>

      <Button
        onClick={onClick}
        text="Try Again"
        type="default"
        className="col-4 mt-1"
      />
    </div>
  );
}
export default Limitation;
