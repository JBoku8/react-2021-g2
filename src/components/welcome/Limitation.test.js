import { render, screen } from '@testing-library/react';
import Limitation from './Limitation';

describe('Limitation component test', () => {
  it('should render limitation component with message', () => {
    render(<Limitation message="message" />);

    const messageElement = screen.getByText('message');
    expect(messageElement).toBeInTheDocument();
  });

  it('should render limitation component with button and execute onclick', () => {
    const onClick = jest.fn();
    render(<Limitation onClick={onClick} />);

    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toBeInTheDocument();
    buttonElement.click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
