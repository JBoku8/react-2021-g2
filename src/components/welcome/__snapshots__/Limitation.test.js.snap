// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Limitation snapshots renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row flex-column"
      >
        <h3
          class="text-danger col-4 mt-3"
        />
        <button
          class="btn mx-1 default col-4 mt-1"
          type="button"
        >
          Try Again
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row flex-column"
    >
      <h3
        class="text-danger col-4 mt-3"
      />
      <button
        class="btn mx-1 default col-4 mt-1"
        type="button"
      >
        Try Again
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
